### PR TITLE
JS part 2(advanced): toast added + cleanup handled + error styling added

### PR DESCRIPTION
**Pull request by FarazPourebrahim**

### CODE REVIEW

File: challenge_01/script.js [hunk 1]
```diff
@@ -9,55 +9,98 @@ formElements.radioBtnSR = document.querySelector("#radio-btn-SR");
...
+        }
+        errorNode.classList.add("problem");
+        showMessage(message,"error")
         console.error(message);
-        shouldReturnEarly = true;
+        errorNode.onchange = () => {errorNode.classList.remove("problem");}
+    }
+
+    function cleanup(){
+        formElements.firstNameInput.value = formElements.lastNameInput.value = formElements.emailInput.value =  formElements.messageTextarea.value = null;
```
Comment (atareversei) [inline challenge_01/script.js]: This is not java😂
The convention in JavaScript is to do it one by one.

File: challenge_01/script.js [hunk 2]
```diff
@@ -9,55 +9,98 @@ formElements.radioBtnSR = document.querySelector("#radio-btn-SR");
 formElements.messageTextarea = document.querySelector("#message-textarea");
 formElements.consentCheckBox = document.querySelector("#consent-checkbox");
 
+const toast = document.querySelector("#toast");
+const toastText = document.querySelector("#toast-text");
+const toastExit = document.querySelector("#toast-exit");
```
Comment (atareversei) [inline challenge_01/script.js]: You could do the same thing that you did with formElements.

File: challenge_01/index.html [hunk 3]
```diff
@@ -14,6 +14,11 @@
     <script src="script.js" defer></script>
 </head>
 <body>
+<div  id="toast" class="toast-wrapper hidden">
+        <span id="toast-exit" class="exit">×</span>
+        <p id="toast-text" class="toast-text">Lorem ipsum whatever</p>
+</div>
+
```
Comment (atareversei) [inline challenge_01/index.html:21]: It is better to take your toast out of the document flow. Right now, when a notification appears, the whole page jumps and that is bad UX.
By taking it out of the flow, I mean setting a display absolute or fixed on it, depending on the situation.

File: challenge_01/script.js [hunk 4]
```diff
@@ -9,55 +9,98 @@ formElements.radioBtnSR = document.querySelector("#radio-btn-SR");
...
     // disabling the default submit and refresh
     event.preventDefault();
 
 
-    let shouldReturnEarly = false;
     //checking for wrong inputs
     if (!formElements.firstNameInput.value){
-        formErrorHandler("please enter first name");
+        formErrorHandler("please enter first name",formElements.firstNameInput);
+        return;
```
Comment (atareversei) [inline challenge_01/script.js]: You have repeated `return` 9 times. What happens if the requirements change? You have to edit 9 places in your code and this is the root of most bugs.
You can take a boolean flag and while calling `formErrorHandler`, you can flip that boolean if any error occurs.

File: challenge_01/script.js [hunk 5]
```diff
@@ -9,55 +9,98 @@ formElements.radioBtnSR = document.querySelector("#radio-btn-SR");
...
+            errorNode = errorNode.parentNode;
+        }
+        if (errorNode.type === "checkbox"){
+            errorNode = errorNode.parentNode;
+        }
+        errorNode.classList.add("problem");
+        showMessage(message,"error")
         console.error(message);
-        shouldReturnEarly = true;
+        errorNode.onchange = () => {errorNode.classList.remove("problem");}
```
Comment (atareversei) [inline challenge_01/script.js]: `errorNode.onInput` is what you are probably looking for.

**Review by atareversei — APPROVED** [review]
Good Work 🔥
Your JS code could be improved architecturally, but I guess this code review is not the place for that otherwise you had to kind of implement a sophisticated form handler library yourself, so yeah no worries😂
And for the CSS part, try to use logical properties everywhere.
width -> inline-size
top -> inset-block-start
padding-left -> padding-inline-start (or padding-inline-end depending on the situation)
height -> block-size
margin-top -> margin-block-start
border-bottom -> border-block-end
padding-left + padding-right -> padding-inline
top + bottom -> inset-block
and so on, you get the idea.

File: challenge_01/script.js (hunk 1 above)
Comment (FarazPourebrahim) [reply challenge_01/script.js]: Noted 😆

File: challenge_01/script.js (hunk 4 above)
Comment (FarazPourebrahim) [reply challenge_01/script.js]: My bad.
I messed up in one part of code and tried to get back and got wrong version restored which still had the `return`s .

**Comment by FarazPourebrahim** [comment]
Thank you for the reviews.
I'm learning a lot from them 🙌

**Pull request merged**